src/claimtrie.cpp

```diff
@@ -2154,6 +2159,7 @@ bool CClaimTrieCache::incrementBlock(insertUndoType& insertUndo, claimQueueRowTy
         {
             CClaimValue claim;
             assert(removeClaimFromTrie(itEntry->name, itEntry->outPoint, claim, true));
+            claimsToDelete.insert(claim);
```

Comment:
This is empty claim, assert points to nothing in release build, i.e. removeClaimFromTrie will not get called.

Reply:
I don't see it. Makes me confusing now.

Reply:
It looks like the claim parameter is both an in and an out parameter, but you're not passing a valid one in. It appears that way to me also because the internal removeClaim does a swap, i.e, it expects valid data to be there.

Reply:
It looks like the no-NDEBUG requirement hasn't yet changed upstream either. I'm surprised, as using asserts to accomplish real work is generally considered an anti-pattern.